### DISCUSSION TITLE
Smartshift - disable ability to charge during demand window

### DISCUSSION THREAD

**Discussion by herbi3** [amberelectric/public-api#134] [Ideas]
It would be great to have the ability to prevent the battery from charging during the demand window for customers that have a demand charge.

Example, I believe my demand charge window is 4pm-9pm. My battery from smartshift started charging at 4pm until 4:30 today. This caused our demand charge to rise to $24.88 for this month. This can get fairly expensive if the batteries continued charging

Without the smartshift api just yet, we can't control this ourselves - is there a way to request this as something fairly high in the list of priorities for Amber please?

**Comment by madpilot** [comment]
It's pencilled in for Q3 (but as with these things might change).

**Reply by herbi3** [reply]
Is that for both the API and limiting the ability to charge during the demand window? Or just the API

**Reply by madpilot** [reply]
Sorry, API is Q3.

We done other pricing stuff to sort out first that'll happen shortly, then hopefully we can slot demand in after that.

**Comment by freman** [comment]
I'd be content with the ability to just disable and enable smartshift via an API, yesterday it was like "Oh, price looks ok at 8pm, we'll run you off the grid for an hour"... no please don't that's, in the demand window.

**Comment by madpilot** [comment]
We are kicking off making Smartshift demand tariff aware now, so it _might_ drop this quarter.
